connectors/src/connectors/notion/lib/notion_api.ts

```diff
@@ -380,12 +380,30 @@ export async function isAccessibleAndUnarchived(
     } catch (e) {
       if (APIResponseError.isAPIResponseError(e)) {
         if (NOTION_RETRIABLE_ERRORS.includes(e.code)) {
-          const waitTime = 500 * 2 ** tries;
+          let waitTime = 500 * 2 ** tries;
+          let usingHeader = false;
+
+          try {
+            // Trying to respect the rate limit sent back by notion api
+            // See: https://developers.notion.com/reference/request-limits
+            const responseHeaders = e.headers as Record<string, string>;
+            if (responseHeaders["Retry-After"]) {
+              const retryAfter = parseInt(responseHeaders["Retry-After"], 10);
+              if (retryAfter > 0) {
```

Comment:
`nit` Given that it comes from the outside you can also use `!isNaN(retryAfter)`. Since `parseInt` is pretty good at returning `NaN` 🙃.

Reply:
<img width="93" alt="image" src="https://github.com/user-attachments/assets/c3a51276-e302-4e4b-8e62-d608852636ea">
I don't think it's needed then, right ?

Reply:
Tricky 🤯  I guess the more you are on the explicit side the better 😉.

Reply:
"mouais" :) but i'll do it to make you happy :)